crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -156,8 +159,26 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
 
     let mut typos_found = false;
     let mut errors_found = false;
-    for path in args.path.iter() {
+
+    let stdin_paths = if args.stdin_paths {
+        Some(
+            io::stdin()
+                .lines()
+                .map(|res| res.map(PathBuf::from))
+                .collect::<Result<_, _>>()
+                .with_code(proc_exit::sysexits::IO_ERR)?,
+        )
+    } else if args.stdin_paths0 {
+        Some(unix_read_paths_from_stdin()?)
+    } else {
+        None
+    };
+
+    for path in stdin_paths.as_ref().unwrap_or(&args.path) {
         let cwd = if path == std::path::Path::new("-") {
+            if args.is_using_stdin_paths() {
+                proc_exit::sysexits::NO_INPUT.ok()?;
```

Comment:
This will likely need an error message to go along with a failing exit code

Reply:
I don't know how to add error messages. :D Please help.

Reply:
`proc_exit::sysexits::NO_INPUT.with_message(...)`

Reply:
TIL. But I decided that when passing through stdin `-` should be treated like any other path.